openmc/mgxs_library.py

```diff
@@ -379,200 +447,437 @@ def tabular_legendre(self, tabular_legendre):
-        # throughout this code.
-        if self._representation is 'isotropic':
-            shape_vec = (self._energy_groups.num_groups,)
-            shape_mat = (self._energy_groups.num_groups,
-                         self._energy_groups.num_groups)
-        elif self._representation is 'angle':
-            shape_vec = (self._num_polar, self._num_azimuthal,
-                         self._energy_groups.num_groups)
-            shape_mat = (self._num_polar, self._num_azimuthal,
-                         self._energy_groups.num_groups,
-                         self._energy_groups.num_groups)
-
-        # Begin by checking the case when chi has already been given and thus
-        # the rules for filling in nu_fission are set.
         if self._use_chi is not None:
             if self._use_chi:
-                shape = shape_vec
+                check_value('nu_fission shape', npnu_fission.shape,
+                            [self.vector_shape])
             else:
-                shape = shape_mat
-            if nu_fission.shape != shape:
-                msg = "Invalid Shape of Nu_fission!"
-                raise ValueError(msg)
+                check_value('nu_fission shape', npnu_fission.shape,
+                            [self.matrix_shape])
         else:
-            # Get shape of nu_fission so we can figure if we need chi or not
-            if nu_fission.shape == shape_vec:
+            check_value('nu_fission shape', npnu_fission.shape,
+                        [self.vector_shape, self.matrix_shape])
+            # Find out if we have a nu-fission matrix or vector
+            # and set a flag to allow other methods to check this later.
+            if npnu_fission.shape == self.vector_shape:
                 self._use_chi = True
-                shape = shape_vec
-            elif nu_fission.shape == shape_mat:
-                self._use_chi = False
-                shape = shape_mat
             else:
-                msg = "Invalid Shape of Nu_fission!"
-                raise ValueError(msg)
+                self._use_chi = False
+
+        self._nu_fission = npnu_fission
+        if np.sum(self._nu_fission) > 0.0:
+            self._fissionable = True
+
+    def set_total_mgxs(self, total, nuclide='total', xs_type='macro'):
+        """This method allows for an openmc.mgxs.TotalXS or
+        openmc.mgxs.TransportXS to be used to set the total cross section
+        for this XSdata object.
+
+        Parameters
+        ----------
+        total: openmc.mgxs.TotalXS or openmc.mgxs.TransportXS
+            MGXS Object containing the total or transport cross section
+            for the domain of interest.
+        nuclide : str
+            Individual nuclide (or 'total' if obtaining material-wise data)
+            to gather data for.  Defaults to 'total'.
+        xs_type: {'macro', 'micro'}
+            Provide the macro or micro cross section in units of cm^-1 or
+            barns. Defaults to 'macro'.
+
+        """
```

Comment:
I'd imagine most people will not manually be using these methods. Can you add a "See also" section on the docstring that references `openmc.mgxs.Library.create_mg_library` and `openmc.mgxs.Library.get_xsdata`? Same applies to all the other `set_*` methods.